hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/upgrade/FinalizationStateManagerImpl.java

```diff
@@ -172,6 +188,68 @@ public FinalizationCheckpoint getFinalizationCheckpoint() {
     return currentCheckpoint;
   }
 
+  @Override
+  public void reinitialize(Table<String, String> newFinalizationStore)
+      throws IOException {
+    checkpointLock.writeLock().lock();
+    try {
+      newFinalizationStore.close();
+      this.finalizationStore = newFinalizationStore;
```

Comment:
Shouldn't `this.finalizationStore` be closed instead of the new one?

Reply:
Yes, good catch. I would have thought this would crash the test, but I guess RocksDB didn't free the resources before the snapshot install process read the layout version key. I will fix this in #3534, since that PR builds off this one and this PR is somewhat obsolete now.